deepspeed/runtime/activation_checkpointing/checkpointing.py

```diff
@@ -311,6 +311,44 @@ def get_full_inputs(tensors, device=None):
     return tuple(inputs)
 
 
+def move_to_device(item, device):
+    if torch.is_tensor(item):
+        return item.to(device)
+    elif isinstance(item, list):
+        return [move_to_device(v, device) for v in item]
+    elif isinstance(item, tuple):
+        return tuple([move_to_device(v, device) for v in item])
+    elif isinstance(item, dict):
+        return {k: move_to_device(v, device) for k, v in item.items()}
+    else:
+        return item
+
+
+def filter_tensor_values(all_values):
+    tensor_values = [v for v in all_values if torch.is_tensor(v)]
+    non_tensor_values = [v for v in all_values if not torch.is_tensor(v)]
+    tensor_flags = [torch.is_tensor(v) for v in all_values]
+    if type(all_values) is tuple:
+        return tuple(tensor_values), tuple(non_tensor_values), tuple(tensor_flags)
+    return tensor_values, non_tensor_values, tensor_flags
+
+
+def merge_tensor_values(tensor_values, non_tensor_values, tensor_flags):

```

Comment:
Not relevant to this PR, but this seems like a pattern that could be reused in other places? I think in the future these merge/filter routines could be generalized and reused to do things like split/merge mixed-type lists of tensors for bucketing.